noxfile.py

```diff
@@ -40,19 +40,66 @@ def set_environment_variables(env_dict, session):
 def run_pybamm_requires(session):
     """Download, compile, and install the build-time requirements for Linux and macOS: the SuiteSparse and SUNDIALS libraries."""  # noqa: E501
     set_environment_variables(PYBAMM_ENV, session=session)
+    force_rebuild = "--force" in session.posargs
+    SUNDIALS_LIBS = [
+        "libsundials_idas",
+        "libsundials_sunlinsolklu",
+        "libsundials_sunlinsoldense",
+        "libsundials_sunlinsolspbcgs",
+        "libsundials_sunlinsollapackdense",
+        "libsundials_sunmatrixsparse",
+        "libsundials_nvecserial",
+        "libsundials_nvecopenmp"
+    ]
+
+    KLU_LIBS = [
+        "libsuitesparseconfig",
+        "libklu",
+        "libamd",
+        "libcolamd",
+        "libbtf"
+    ]
```

Comment:
@agriyakhetarpal, I think we should check for the existing installation in `install_KLU_Sundials.py` itself? We recommend developers to use `nox` but we also have documentation for doing stuff without `nox`, so it might be better to use `nox` just as a wrapper for these scripts.

IIRC the checks are also present in `install_odes.py` (and not the `nox` session of `install_odes.py`).

Reply:
I would recommend doing these checks with `nox` rather than inside the script because `nox` is meant to ease the installation of the build-time requirements both for newer contributors and for experienced developers. The rationale behind my suggestion is that not only does `nox` provide a powerful command-line interface and can abstract away the extra code required to implement a command-line argument parser in a Python script (not that doing that would take time, though), it is also our current "recommended" method of installing from source in the installation instructions. Shall someone run `install_KLU_Sundials.py` and clone the `pybind11` repository manually, it should be assumed that they know what they are doing and the script should _recompile_ SuiteSparse and SUNDIALS as usual even if both components already exist on the system.

However, `pybamm_install_odes` is provided as an entry point for use in user installations. This is why we should skip adding a `nox` session for it – we don't bundle the noxfile in the source distribution or in the wheels anyway, plus for developers this is redundant (`pip install pybamm[odes]` now does the job once `SUNDIALS_INST` is correctly set). Therefore, `install_odes.py` should keep those checks inside itself since it is a standalone application that is shipped with PyBaMM.

Note: yes, the checks are inside `install_odes.py` but they are not complete; just one SUNDIALS-specific file is checked, i.e., `libsundials_idas.so`, compared to the list of files I have included here. We should update that soon (@arjxn-py, could you do that in #3417?)

<hr>

P.S. going forward in the coming months in the aftermath of adding `pyproject.toml`, the changes here will not matter since steps can be taken to eliminate the dependence on `pybamm-requires` completely wherein a combination of a robust caching mechanism inside the script with hashes and the use of CCache can be devised to speed up and simplify the installation for both users and developers.

Reply:
> could you do that in #3417?

Yes sure, hope it's fine to do it after completing #3475